tests/testthat/test-variable-value.R

```diff
@@ -0,0 +1,178 @@
+                          showSelected=bases.per.problem,
+                          xend=problemEnd, yend=problem.i),
+                      size=5,
+                      data=data.frame(problems, sample.id="problems"))+
+         geom_text(aes(200, 5,
+                       label=paste("problem size", bases.per.problem),
+                       showSelected=bases.per.problem),
+                   data=data.frame(sizes, sample.id="problems"))+
+         theme_bw()+
+         theme(panel.margin=grid::unit(0, "cm"))+
+         facet_grid(sample.id ~ .),
+
+       title="viz with for loop",
+       
+       sizes=ggplot()+
+         ggtitle("select problem size")+
+         geom_point(aes(bases.per.problem, problems,
+                        clickSelects=bases.per.problem),
+                    size=10,
+                    data=sizes),
+
+       peaks=ggplot()+
+         ggtitle("select number of peaks")+
+         geom_text(aes(1, 3, label=problem.name,
+                       showSelected=problem.name),
+                   data=problems))
+
+pp.list <- split(peak.problems, peak.problems$problem.name)
+s.list <- split(samples, samples$problem.name)
+p.list <- split(peaks, peaks$problem.name)
+
+for(problem.name in names(p.list)){
+  s.name <- paste0(problem.name, "peaks")
+  p <- p.list[[problem.name]]
+  p[[s.name]] <- p$peaks
+  pp <- pp.list[[problem.name]]
+  pp[[s.name]] <- pp$peaks
+  s <- s.list[[problem.name]]
+  s[[s.name]] <- s$peaks
+  p$bases.per.problem <- pp$bases.per.problem[1]
+  viz.for$problems <- viz.for$problems+
+    geom_segment(aes_string("peakStart", "problem.i",
+                            showSelected=s.name,
+                            clickSelects="problem.name",
+                            showSelected2="bases.per.problem",
+                            xend="peakEnd", yend="problem.i"),
+                 data=data.frame(pp, sample.id="problems"),
+                 size=10,
+                 color="deepskyblue")+
+    geom_segment(aes_string("peakStart", "0",
+                            showSelected=s.name,
+                            clickSelects="problem.name",
+                            showSelected2="bases.per.problem",
+                            xend="peakEnd", yend="0"),
+                 data=s,
+                 size=10,
+                 color="deepskyblue")
+  viz.for$peaks <- viz.for$peaks+
+         geom_point(aes_string("peaks", "peaks",
+                               showSelected="problem.name",
+                               showSelected2="bases.per.problem",
+                               clickSelects=s.name),
+                    size=10,
+                    data=p)
+}
+
+##animint2dir(viz.for, "problem-peaks-for")
+
+test_that("some chunks are not downloaded", {
+  info <- animint2HTML(viz.for)
+  node.set <-
+    getNodeSet(info$html, '//td[@class="downloaded"]')
+  value.vec <- sapply(node.set, xmlValue)
+  expect_true("0" %in% value.vec)
+  expect_match(value.vec,  "[01]")
+})
```

Comment:
this test makes sure that some chunks are not downloaded at first (0) and that those which are downloaded only have 1 chunk.